Unhelpful NullPointerException when no JDK is installed

Apparently gradle fails with an entirely unhelpful error message

> java.lang.NullPointerException (no error message)

when no JDK is installed, only a JRE. See elki-project/elki/issues/91

Gradle should instead report that there is no Java JDK installed.

Relevant backtrace (full output in https://github.com/elki-project/elki/files/6716841/output.log ):
```
#8 34.82 Caused by: java.lang.NullPointerException
#8 34.82        at org.gradle.api.internal.tasks.compile.JdkTools$DefaultIncrementalAwareCompiler.getStandardFileManager(JdkTools.java:136)
#8 34.82        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.createCompileTask(JdkJavaCompiler.java:67)
#8 34.82        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:54)
#8 34.82        at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:40)
```

--- a/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkTools.java
+++ b/subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/JdkTools.java
@@ -103,7 +103,14 @@ public class JdkTools {
             if (isJava9Compatible) {
                 clazz = isolatedToolsLoader.loadClass("javax.tools.ToolProvider");
                 try {
-                    return (JavaCompiler) clazz.getDeclaredMethod("getSystemJavaCompiler").invoke(null);
+                    JavaCompiler compiler = (JavaCompiler) clazz.getDeclaredMethod("getSystemJavaCompiler").invoke(null);
+                    if (compiler == null) {
+                        // We were trying to load a compiler in our process so Jvm.current() is the correct one to blame.
+                        throw new IllegalStateException("Java compiler is not available. Please check that "
+                            + Jvm.current().getJavaHome().getAbsolutePath()
+                            + " contains a valid JDK installation.");
+                    }
+                    return compiler;
                 } catch (IllegalAccessException | NoSuchMethodException | InvocationTargetException e) {
                     cannotCreateJavaCompiler(e);
                 }

